File: devBuilder/src/main/java/net/medrag/devBuilder/controller/RController.java
package net.medrag.devBuilder.controller;

import net.medrag.devBuilder.DevBuilderApplication;
import net.medrag.devBuilder.model.Request;
import net.medrag.devBuilder.model.StartUpData;
import net.medrag.devBuilder.service.Joker;
import net.medrag.devBuilder.service.Processor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import javax.validation.Valid;

/**
 * {@author} Stanislav Tretyakov
 * 10.10.2019
 */
@RestController
@RequestMapping("/devBuilder")
public class RController {

    private Processor processor;
    private Joker joker;

    @Autowired
    public RController(Processor processor, Joker joker) {
        this.processor = processor;
        this.joker = joker;
    }

    @GetMapping(value = "/getDeveloper/{taskId}", produces = MediaType.TEXT_XML_VALUE)
    public String getDev(@PathVariable("taskId") String taskId) {
        return processor.process(taskId);
    }

    @PostMapping(value = "/setSkillsCount", produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
    public ResponseEntity<Request> setServicesAmount(@RequestBody @Valid Request request) {
        return ResponseEntity.ok(processor.setServicesAmount(request));
    }

    @DeleteMapping("/removeRequest")
    public ResponseEntity<Request> removeRequest(@RequestParam String ending) {
        return ResponseEntity.ok(processor.removeRequest(ending));
    }

    @GetMapping(value = "/getStartUpData", produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
    public ResponseEntity<StartUpData> getStartUpData() {
        return ResponseEntity.ok(processor.getStartUpData());
    }

    /**
     * Getting a joke
     *
     * @return - fresh random joke
     */
    @GetMapping(path = "/getJoke", produces = MediaType.TEXT_PLAIN_VALUE)
    public String getJoke() {
        return joker.getJoke();
    }

    @GetMapping("/shutdown")
    public void shutdown() {
        DevBuilderApplication.shutDown();
    }


}

File: ktapp-parent/ktapp/src/main/kotlin/net/medrag/ktapp/controller/RController.kt
package net.medrag.ktapp.controller

import mu.KotlinLogging
import net.medrag.ktapp.context.ProcessContextParameter
import net.medrag.ktapp.context.withMdc
import net.medrag.ktapp.service.SuspendService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RestController
import java.util.*

/**
 * @author Stanislav Tretiakov
 * 17.07.2023
 */
@RestController
class RController(
    private val suspendService: SuspendService
) {

    @GetMapping("/get/{value}")
    suspend fun processGet(@PathVariable value: String): String {
        return withMdc {
            log.info { "controller... $value" }
            suspendService.process(value)
        }
    }

    @GetMapping("/getit")
    suspend fun processGet(): String {
        return withMdc({ it.enrich(ProcessContextParameter.SOME_DATA, "QQQ") }) {
            log.info("get it...")
            UUID.randomUUID().toString()
        }
    }

    companion object {
        private val log = KotlinLogging.logger { }
    }
}